src/udp_wrap.cc

```diff
@@ -324,6 +328,43 @@ void UDPWrap::DropMembership(const FunctionCallbackInfo<Value>& args) {
   SetMembership(args, UV_LEAVE_GROUP);
 }
 
+void UDPWrap::SetSourceMembership(const FunctionCallbackInfo<Value>& args,
+                            uv_membership membership) {
+  UDPWrap* wrap;
+  ASSIGN_OR_RETURN_UNWRAP(&wrap,
+                          args.Holder(),
+                          args.GetReturnValue().Set(UV_EBADF));
+
+  CHECK_EQ(args.Length(), 3);
+
+  node::Utf8Value sourceAddress(args.GetIsolate(), args[0]);
+  node::Utf8Value groupAddress(args.GetIsolate(), args[1]);
+  node::Utf8Value iface(args.GetIsolate(), args[2]);
+
+  const char* iface_cstr = *iface;
+  if (args[2]->IsUndefined() || args[2]->IsNull()) {
```

Comment:
I'm not sure if we want to support `null` here, as that is generally intentional, where `undefined` is not. You can also move `node::Utf8Value iface(args.GetIsolate(), args[2]);` from above into the `else` branch so we don't compute it for no reason.

Reply:
This bit of code is a bit tricky. I've tried to follow your suggestion, inverting the check and then assigning the pointer only if its a real thing.

As stated in my earlier comment, the function is based off of an existing one with a similar interface. Seemingly, it is permissable to explicitly call these functions with `null`, which will trigger a call down into libuv with `nullptr` effectively a no interface defined. 

I'm happy to change this but would really want the dgram API to be consistent across multicast functions. So any changes would probably need to be unilateral.